wordpress.org/public_html/wp-content/mu-plugins/pub/wporg-seo/canonical.php

```diff
@@ -0,0 +1,98 @@
+<?php
+namespace WordPressdotorg\SEO\Canonical;
+/**
+ * Adds canonical-related functionality.
+ * @see https://core.trac.wordpress.org/ticket/18660
+ */
+
+/**
+ * Outputs a <link rel="canonical"> on most pages.
+ */
+function rel_canonical_link() {
+	if ( $url = get_canonical_url() ) {
+		printf(
+			'<link rel="canonical" href="%s">' . "\n",
+			esc_url( $url )
+		);
+	}
+}
+add_action( 'wp_head', __NAMESPACE__ . '\rel_canonical_link' );
+add_action( 'login_head',  __NAMESPACE__ . '\rel_canonical_link' );
+
+remove_action( 'wp_head', 'rel_canonical' );
+
+/**
+ * Get the current Canonical URL.
+ */
+function get_canonical_url() {
+	global $wp, $wp_rewrite;
+
+	$queried_object = get_queried_object();
+	$url = false;
+
+	if ( is_tax() || is_tag() || is_category() ) {
+		$url = get_term_link( $queried_object );
+	} elseif ( is_singular() ) {
+		$url = get_permalink( $queried_object );
+	} elseif ( is_search() ) {
+		$url = home_url( 'search/' . urlencode( get_query_var( 's' ) ) . '/' );
+	} elseif ( is_author() ) {
+		// On WordPress.org get_author_posts_url() returns profile.wordpress.org links. Build it manually.
+		$url = home_url( 'author/' . $queried_object->user_nicename . '/' );
+	} elseif ( is_post_type_archive() ) {
+		$url = get_post_type_archive_link( $queried_object->name );
+	} elseif ( is_home() ) {
+		$url = get_post_type_archive_link( 'post' );
+	} elseif ( is_front_page() ) {
+		$url = home_url( '/' );
+	} elseif ( is_date() ) {
+		if ( is_day() ) {
+			$url = get_day_link( get_query_var('year'), get_query_var('monthnum'), get_query_var('day') );
+		} elseif ( is_month() ) {
+			$url = get_month_link( get_query_var('year'), get_query_var('monthnum') );
+		} elseif ( is_year() ) {
+			$url = get_year_link( get_query_var('year') );
+		}
+	}
```

Comment:
If I understand correctly, this will output a canonical meta tag for search results and archives. However, these content types are also no-indexed in robots.php.

Ideally the canonical url meta would no be output on no-index pages, and Google has previously indicated the presence of a canonical URL may take precedence over the no-index: https://www.searchenginejournal.com/google-dont-mix-noindex-relcanonical/262607/

Reply:
If you feel strongly about that, you should report that upstream to the Meta Trac, I don't think that's something I'm going to fix here, considering I'm just trying to combine all the existing canonical and noindexing code we have.

Reading that article though, it suggests to me that if rel=canonical was directed to a noindex page, but the others weren't, then it may ignore the noindex.

If I had my way, no content on WordPress.org would be noindex.